w3af/core/data/context/context/html.py

```diff
@@ -133,12 +156,18 @@ class HTMLAttrQuoteGeneric(BaseContext):
     JS_PATTERN = re.compile('^ *javascript:', re.IGNORECASE)
     VB_PATTERN = re.compile('^ *vbscript:', re.IGNORECASE)
 
-    def __init__(self, payload, attr_name, attr_value):
+    # What about &NewLine;, &Tab;, etc?
+    JS_PROTOCOLS = {'javascript:', 'javascript&colon;',
```

Comment:
> Browser expects <a href="javascript:foo(2)"> not <a href="javascript&colon;foo(2)">

Yes, but browser will unescape all html entities before processing uri. So `javascript&colon;foo(2)` transforms to `javascript:foo(2)` and successfully works. That is why some WAFs can be bypassed by using html5 entities, e.g. `<a href="javascript&Tab;&colon;foo(2)">`